src/extensionOnPrem.ts

```diff
@@ -0,0 +1,52 @@
+import * as vscode from "vscode";
+import {
+  deactivate as requestDeactivate,
+  initBinary,
+} from "./binary/requests/requests";
+import { registerCommands } from "./commandsHandler";
+import { provideHover } from "./hovers/hoverHandler";
+import { registerStatusBar, setDefaultStatus } from "./statusBar/statusBar";
+import { setBinaryRootPath } from "./binary/paths";
+import { setTabnineExtensionContext } from "./globals/tabnineExtensionContext";
+
+import installAutocomplete from "./autocompleteInstaller";
+import { handleSelection, notifyBinaryAboutWorkspaceChange } from "./extension";
+
+export async function activate(
+  context: vscode.ExtensionContext
+): Promise<void> {
+  console.log("*****THIS IS AN ONPREM BUILD****");
+  void initStartup(context);
+  handleSelection(context);
+  registerStatusBar(context);
```

Comment:
i would override this function and and create an onprem `registerStatusBar` function instead of modifying the `updateStatus` function